### PR TITLE
#29 profile dob and seeder

### PR DESCRIPTION
**Pull request by dpslwk**
This should cover adding the DoB field and the DB seeder needs for #29

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -25,11 +27,14 @@ public function boot()
...
     public function register()
     {
-        //
         $this->app->singleton(PasswordStore::class, function ($app) {
             $passwordStoreManager = new PasswordStoreManager($app);
 
             return $passwordStoreManager->driver();
         });
+
+        $this->app->singleton(FakerGenerator::class, function () {
```
Comment (geeksareforlife) [inline app/Providers/AppServiceProvider.php]: Why is this needed? The user seeder is working seemingly exactly like the new profile seeder already.

Comment (dpslwk) [inline app/Providers/AppServiceProvider.php]: cause with out it you get American addresses and phone numbers, this overriders the doctrine US default

File: app/HMS/Entities/Profile.php [hunk 2]
```diff
@@ -262,4 +265,23 @@ public function setContactNumber(string $contactNumber)
...
+    /**
+     * @return Carbon|null
+     */
+    public function getDateOfBirth()
+    {
+        return $this->dateOfBirth;
+    }
+
+    /**
+     * @param Carbon $joinDate
```
Comment (geeksareforlife) [inline app/HMS/Entities/Profile.php]: think you mean $dateOfBirth here

File: app/HMS/User/ProfileManager.php [hunk 3]
```diff
@@ -20,14 +22,21 @@ class ProfileManager
     protected $userRepository;
 
     /**
+     * @var MetsRepository
```
Comment (geeksareforlife) [inline app/HMS/User/ProfileManager.php]: MetaRepository

File: database/factories/ProfileFactory.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    return [
+        'user' => isset($attributes['user']) ? $attributes['user'] : null,
+        'unlockText' => 'Welcome ' . $attributes['user']->getFirstName(),
+        'creditLimit' => \Meta::get('member_credit_limit'),
+        'address1' => $faker->streetAddress,
+        'addressCity' => $faker->city,
+        'addressCounty' => $faker->county,
+        'addressPostcode' => $faker->postcode,
+        'contactNumber' => $faker->phoneNumber,
+        'dateOfBirth' => Carbon::instance($faker->dateTimeBetween($startDate = '-18 years', $endDate = '-16 years')),
```
Comment (geeksareforlife) [inline database/factories/ProfileFactory.php]: Think the range here is wrong - 100 to 18 maybe?

**Comment by geeksareforlife** [comment]
Works fine
